### ISSUE TITLE
Sorting in content delivery API not working

### ISSUE THREAD

**Issue by mhcom** [storyblok/storyblok#811]
**The issue occurs while working with:** 
- [ ] *app.storyblok.com (CMS - Interface)*   
- [x] *api.storyblok.com (CMS - Content Delivery API)*
- [ ] *mapi.storyblok.com (CMS - Management API)*
- [ ] *capi.storyblok.com (Commerce - API)*     
- [ ] *Commerce - Interface*   
- [ ] *Other:* 

**Current behavior:**
The sorting in the content delivery API (https://www.storyblok.com/docs/api/content-delivery#examples/ordering/sort-by-content-attribute) doesn't work reliably.

We have 100s of stories, across dozens of folders, which we want to **sort by a content attribute**. However the result we get from Storyblok is not correctly sorted by this field.

**Expected behavior:**
Sorting works, even when stories are stored across folders.

Furthermore it needs to be ensured and tested that sorting works when filters are applied to the search query, eg. filtering by a multi-select value of a story.

**Comment by VojislavVlasic** [comment]
Hi @mhcom can you please provide us with an example for this situation?

**Comment by mhcom** [comment]
Hi @VojislavVlasic we provided examples to your support team for a long while. There's currently a support case still pending. We had prepared a test case for them here: https://app.storyblok.com/#/me/spaces/110390/stories/0/0/index/119003897
<img width="173" alt="Screenshot 2022-10-05 at 09 33 16" src="https://user-images.githubusercontent.com/5443217/193967956-ec270698-02f2-457e-9c89-d23ae3a92c25.png">

- All of these stories and folders have been manually sorted in Storyblok via the _move_ functionality.
- Each folder has a root story assigned.

Expectation is that we can retrieve the stories and root stories via the content delivery API in the order, in which they are sorted in Storyblok:
1. Sorting Test Page 1
2. Sorting Test Folder 2 Root
3. Sorting Test Folder 4 Root
4. Sorting Test Folder 5 Root

Unfortunately there seems to be no way of retrieving these stories this way from the content delivery API.
